acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java

```diff
@@ -0,0 +1,160 @@
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import com.google.common.io.MoreFiles;
+import com.google.common.io.RecursiveDeleteOption;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.awaitility.Awaitility;
+
+public abstract class Eth2SignerRunner {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final SignerConfiguration signerConfig;
+  private final Path dataPath;
+  private final Properties portsProperties;
+
+  private static final String PORTS_FILENAME = "eth2signer.ports";
+  private static final String HTTP_PORT_KEY = "http-port";
+
+  public static Eth2SignerRunner createRunner(final SignerConfiguration signerConfig) {
+    if (Boolean.getBoolean("acctests.runEth2SignerAsProcess")) {
+      return new Eth2SignerProcessRunner(signerConfig);
+    } else {
+      return new Eth2SignerThreadRunner(signerConfig);
+    }
+  }
+
+  public Eth2SignerRunner(final SignerConfiguration signerConfig) {
+    this.signerConfig = signerConfig;
+    this.portsProperties = new Properties();
+
+    if (signerConfig.isDynamicPortAllocation()) {
+      try {
+        this.dataPath = Files.createTempDirectory("acceptance-test");
+      } catch (final IOException e) {
+        throw new RuntimeException(
+            "Failed to create the temporary directory to store the eth2signer.ports file");
+      }
+    } else {
+      dataPath = null;
+    }
+  }
+
+  public void start() {
+    final List<String> params = createCmdLineParams();
+
+    startExecutor(params);
+
+    if (signerConfig.isDynamicPortAllocation()) {
+      loadPortsFile();
+    }
+  }
+
+  protected abstract void startExecutor(final List<String> params);
+
+  public void shutdown() {
+    try {
+      shutdownExecutor();
+    } finally {
+      if (signerConfig.isDynamicPortAllocation()) {
+        try {
+          MoreFiles.deleteRecursively(dataPath, RecursiveDeleteOption.ALLOW_INSECURE);
+        } catch (final IOException e) {
+          LOG.info("Failed to clean up temporary file: {}", dataPath, e);
+        }
+      }
+    }
+  }
+
+  protected abstract void shutdownExecutor();
+
+  public abstract boolean isRunning();
+
+  private List<String> createCmdLineParams() {
+    final String loggingLevel = "DEBUG";
+
+    final List<String> params = new ArrayList<>();
+    params.add("--logging");
+    params.add(loggingLevel);
+    params.add("--http-listen-host");
+    params.add(signerConfig.hostname());
+    params.add("--http-listen-port");
+    params.add(String.valueOf(signerConfig.httpPort()));
+    params.add("--key-store-path");
+    params.add(signerConfig.getKeyStorePath().toString());
+    if (signerConfig.isDynamicPortAllocation()) {
+      params.add("--data-path");
```

Comment:
question, we don't need datapath if we are not using dynamic ports?

Reply:
correct - datapath is where we put our ports.ethsigner file - but if we're using a fixed port, no need to track it - this is historical.